### PR TITLE
Improve fs::PathStartsWith to handle empty strings

### PR DESCRIPTION
**Pull request by asrelo**
In #14194 @sfan5 suggested that `""` does not refer to a proper path, and `fs::PathStartsWith(path, "")` should just return `false`. This is also the case in libraries in other languages where I looked, seems to be common.

The new behavior:
* check early, if `prefix` is empty - return if path is empty or not,
* no special processing for when `path` is empty, the function meets characters in `prefix` and returns false anyway.

Unit test `TestFileSys::testPathStartsWith` is updated for this and no longer fails on Windows.

Also added some comments to `fs::PathStartsWith` because execution flow is a bit confusing.

## To do

This PR is Ready for Review

## How to test

`minetest --run-unittests --test-module TestFileSys` - the unit test doesn't fail on Windows anymore.

### CODE REVIEW

**Comment by asrelo** [comment]
Oh, now it fails on Linux. I think this is because of the "a small change just before push". But I don't have Linux ready to test myself, and I don't know if a push to guess will cause unreasonable spending of Actions time (i don't know GitHub stuff well).

**Comment by rollerozxa** [comment]
> I don't know if a push to guess will cause unreasonable spending of Actions time (i don't know GitHub stuff well).

Actions time is free for public repos and only cause a bit more strain on Microsoft's servers (I'm sure they won't mind :D), feel free to test away using the CI.

**Comment by JosiahWI** [comment]
Would it work to wrap both the full path and the prefix in `std::filesystem::path` (we have C++17 now!) and call `std::equal`?

File: src/filesys.cpp [hunk 1]
```diff
@@ -697,32 +697,43 @@ bool MoveDir(const std::string &source, const std::string &target)
 
 bool PathStartsWith(const std::string &path, const std::string &prefix)
 {
+	if (prefix.size() == 0)
+		return path.size() == 0;
```
Comment (sfan5) [inline src/filesys.cpp:701]:
```suggestion
	if (prefix.empty())
		return path.empty();
```

**Review by sfan5 — APPROVED** [review]
LGTM

**Review by appgurueu — APPROVED** [review]
Code looks good, tests are passing, the fix of the tests makes sense. Also briefly went over the usages of `PathStartsWith`, nothing looks like it would be broken by this.

**Pull request merged**